web/api/health/health_cmdapi.c

```diff
@@ -73,75 +70,86 @@ int web_client_api_request_v1_mgmt_health(RRDHOST *host, struct web_client *w, c
             buffer_strcat(wb, HEALTH_CMDAPI_MSG_AUTHERROR);
             ret = 403;
         } else {
-            while (url) {
-                char *value = mystrsep(&url, "&");
-                if (!value || !*value) continue;
-
-                char *key = mystrsep(&value, "=");
-                if (!key || !*key) continue;
-                if (!value || !*value) continue;
-
-                debug(D_WEB_CLIENT, "%llu: API v1 health query param '%s' with value '%s'", w->id, key, value);
-
-                // name and value are now the parameters
-                if (!strcmp(key, "cmd")) {
-                    if (!strcmp(value, HEALTH_CMDAPI_CMD_SILENCEALL)) {
-                        silencers->all_alarms = 1;
-                        silencers->stype = STYPE_SILENCE_NOTIFICATIONS;
-                        buffer_strcat(wb, HEALTH_CMDAPI_MSG_SILENCEALL);
-                    } else if (!strcmp(value, HEALTH_CMDAPI_CMD_DISABLEALL)) {
-                        silencers->all_alarms = 1;
-                        silencers->stype = STYPE_DISABLE_ALARMS;
-                        buffer_strcat(wb, HEALTH_CMDAPI_MSG_DISABLEALL);
-                    } else if (!strcmp(value, HEALTH_CMDAPI_CMD_SILENCE)) {
-                        silencers->stype = STYPE_SILENCE_NOTIFICATIONS;
-                        buffer_strcat(wb, HEALTH_CMDAPI_MSG_SILENCE);
-                    } else if (!strcmp(value, HEALTH_CMDAPI_CMD_DISABLE)) {
-                        silencers->stype = STYPE_DISABLE_ALARMS;
-                        buffer_strcat(wb, HEALTH_CMDAPI_MSG_DISABLE);
-                    } else if (!strcmp(value, HEALTH_CMDAPI_CMD_RESET)) {
-                        silencers->all_alarms = 0;
-                        silencers->stype = STYPE_NONE;
-                        free_silencers(silencers->silencers);
-                        silencers->silencers = NULL;
-                        buffer_strcat(wb, HEALTH_CMDAPI_MSG_RESET);
-                    }
-                } else {
-                    uint32_t hash = simple_uhash(key);
-                    if (unlikely(silencer == NULL)) {
-                        if (
-                                (hash == hash_alarm && !strcasecmp(key, HEALTH_ALARM_KEY)) ||
-                                (hash == hash_template && !strcasecmp(key, HEALTH_TEMPLATE_KEY)) ||
-                                (hash == hash_chart && !strcasecmp(key, HEALTH_CHART_KEY)) ||
-                                (hash == hash_context && !strcasecmp(key, HEALTH_CONTEXT_KEY)) ||
-                                (hash == hash_host && !strcasecmp(key, HEALTH_HOST_KEY)) ||
-                                (hash == hash_families && !strcasecmp(key, HEALTH_FAMILIES_KEY))
-                                ) {
-                            silencer = create_silencer();
+            uint32_t end = w->total_params;
+            if (end) {
```

Comment:
Code style:
Same as above, you can replace:
```
             if (end) {
                uint32_t i = 0;
                do {
```
with `for (i = 0 ; i < end ; ++i) {`